js/pinnedLinks.js

```diff
@@ -0,0 +1,142 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+ /*globals gUserDatabase, OBJECT_STORE_PREFS, PINNED_LINKS_PREF */
+
+"use strict";
+
+/**
+ * Singleton that keeps track of all pinned links and their positions in the
+ * grid.
+ */
+
+(function(exports) {
+  const gPinnedLinks = {
+    /**
+     * The cached list of pinned links.
+     */
+    _links: [],
+
+    /**
+     * Set the array of pinned links.
+     */
+    setLinks(loadedLinks) {
```

Comment:
Q: why do we take a string to then just convert it to objects? Can't we just take a list of objects?